concordium-std/src/types.rs

```diff
@@ -635,14 +635,23 @@ pub type ReadOnlyCallContractResult<A> = Result<Option<A>, CallContractError<A>>
 pub type TransferResult = Result<(), TransferError>;
 
 /// A type representing the attributes, lazily acquired from the host.
-#[derive(Default)]
+#[derive(Clone, Copy, Default)]
 pub struct AttributesCursor {
     /// Current position of the cursor, starting from 0.
     /// Note that this is only for the variable attributes.
     /// `created_at` and `valid_to` will require.
     pub(crate) current_position: u32,
     /// The number of remaining items in the policy.
     pub(crate) remaining_items:  u16,
+    /// The total number of items. Used for creating new attribute cursors.
+    pub(crate) total_items:      u16,
```

Comment:
Do we need this? Can we not always get it by using
```
total_items = remaining_items + current_position
```
?

Reply:
Yes, we do. `current_position` relates to the number of bytes read _not_ the number of items.